src/test/java/com/mongodb/hibernate/jdbc/MongoResultSetTests.java

```diff
@@ -0,0 +1,170 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.bson.BsonDocument;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class MongoResultSetTests {
+    private static final List<String> FIELDS = List.of("id", "title", "author");
+
+    @Mock
+    private MongoCursor<BsonDocument> mongoCursor;
+
+    private MongoResultSet mongoResultSet;
+
+    @BeforeEach
+    void setUp() {
+        mongoResultSet = new MongoResultSet(mongoCursor, FIELDS);
+    }
+
+    @Nested
+    class CloseTests {
+
+        @BeforeEach
+        void setUp() throws SQLException {
+            mongoResultSet.close();
+        }
+
+        @Test
+        @DisplayName("No-op when 'close()' is called on a closed MongoResultSet")
+        void testNoopWhenCloseStatementClosed() throws SQLException {
+            // when && then
+            assertDoesNotThrow(() -> mongoResultSet.close());
+        }
+
+        @ParameterizedTest(name = "SQLException is thrown when \"{0}\" is called on a closed MongoResultSet")
+        @MethodSource("getMongoResultSetMethodInvocationsImpactedByClosing")
+        void testCheckClosed(String label, ResultSetMethodInvocation methodInvocation) throws SQLException {
+            // when && then
+            var exception = assertThrows(SQLException.class, () -> methodInvocation.runOn(mongoResultSet));
+            assertEquals("MongoResultSet has been closed", exception.getMessage());
+        }
+
+        private static Stream<Arguments> getMongoResultSetMethodInvocationsImpactedByClosing() {
+            return Map.<String, ResultSetMethodInvocation>ofEntries(
+                            Map.entry("next()", ResultSet::next),
+                            Map.entry("wasNull()", ResultSet::wasNull),
+                            Map.entry("getString(int)", rs -> rs.getString(1)),
+                            Map.entry("getBoolean(int)", rs -> rs.getBoolean(1)),
+                            Map.entry("getByte(int)", rs -> rs.getByte(1)),
+                            Map.entry("getShort(int)", rs -> rs.getShort(1)),
+                            Map.entry("getInt(int)", rs -> rs.getInt(1)),
+                            Map.entry("getLong(int)", rs -> rs.getLong(1)),
+                            Map.entry("getFloat(int)", rs -> rs.getFloat(1)),
+                            Map.entry("getDouble(int)", rs -> rs.getDouble(1)),
+                            Map.entry("getBytes(int)", rs -> rs.getBytes(1)),
+                            Map.entry("getDate(int)", rs -> rs.getDate(1)),
```

Comment:
These are all tested here for close, but elsewhere they are not tested for their values (I suggested tracking this somehow). Which methods is this PR intending to implement?

Reply:
It is fine if we implement and test only a subset of such methods in this PR, and create a followup ticket (I would prefer this, because it unblocks parallel implementation faster). Or we can implement and test all methods in this PR. But in this test, we test all methods, and in a different test, we test only some methods - I am ok with either way as long as it is consistent.